### PR TITLE
Update workspaces docs for clarity and add billing and credits info

### PR DESCRIPTION
**Pull request by rosieyohannan**
N&S tasks - workspaces: https://circleci.atlassian.net/browse/DOCSTEAM-118?atlOrigin=eyJpIjoiODRhYjVkZDM5ZGNlNGM0Yjg0OGFmODU5YjNjNzFlMmUiLCJwIjoiaiJ9

Things I've done:

- pull out workspaces into its own page
- linked to new workspaces page from the old place in workflows doc
- added network egress info, how to find usage
- put items back into the TOC in workflows doc so things are easier to find

to do...
I guess I should add info about charges for storage as well as network egress. Is network egress just for runner users, and overages storage costs is for all users @devinmarieb?

### CODE REVIEW

File: jekyll/_cci2/workspaces.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+---
+
+Workflows each have an associated workspace. Workspaces are used to transfer data to downstream jobs as the workflow progresses.
+
+
+Use workspaces to pass along data that is unique to a workflow and is needed for downstream jobs. Workflows that include jobs running on multiple branches may require data to be shared using workspaces. Workspaces are also useful for projects in which compiled data are used by test containers.
+
+For example, a project with a `build` job that builds a jar and saves it to a workspace. The `build` job fans-out into concurrently running test jobs: `integration-test`, `unit-test`, and `code-coverage`, each of which can have access to the jar by attaching the workspace
+
+## Overview
```
Comment (rosieyohannan) [inline jekyll/_cci2/workspaces.md:18]:
```suggestion
## Overview
{: #overview }
```

File: jekyll/_cci2/workspaces.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+Workspaces are additive-only data storage. Jobs can persist data to the workspace. When a workspace is used, data is archived and stored in an off-container store. With each addition to the workspace a new layer is created in the store. Downstream jobs can then attach the workspace to their container filesystem. Attaching the workspace downloads and unpacks each layer based on the ordering of the upstream jobs in the workflow.
+
+![workspaces data flow]( {{ site.baseurl }}/assets/img/docs/workspaces.png)
+
+## Workspace configuration
```
Comment (rosieyohannan) [inline jekyll/_cci2/workspaces.md:24]:
```suggestion
## Workspace configuration
{: #workspace-configuration }
```

File: jekyll/_cci2/workspaces.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+For a live example of using workspaces to pass data between build and deploy jobs, see the [`config.yml`](https://github.com/circleci/circleci-docs/blob/master/.circleci/config.yml) that is configured to build the CircleCI documentation.
+
+For additional conceptual information on using workspaces, caching, and artifacts, refer to the [Persisting Data in Workflows: When to Use Caching, Artifacts, and Workspaces](https://circleci.com/blog/persisting-data-in-workflows-when-to-use-caching-artifacts-and-workspaces/) blog post.
+
+## Workspace expiration
+{: #workspace-expiration }
+
+Workspaces are stored for up to 15 days. Workspaces are not shared between pipelines, and the only way to access a workspace once the workflow has completed is if the workflow is rerun within the 15 day window.
+
+## Workspaces and runner network charges
```
Comment (rosieyohannan) [inline jekyll/_cci2/workspaces.md:90]:
```suggestion
## Workspaces and runner network charges
{: #workspaces-and-runner-network-charges }
```

File: jekyll/_cci2/workspaces.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+- Cloud
+- Server v3.x
+- Server v2.x
+---
+
+Workflows each have an associated `workspace`. Workspaces are used to transfer data to downstream jobs as the workflow progresses.
+
+Use workspaces to pass along data that is unique to a workflow and is needed for downstream jobs. Workflows that include jobs running on multiple branches may require data to be shared using workspaces. Workspaces are also useful for projects in which compiled data are used by test containers.
+
+For example, a project with a `build` job that builds a jar and saves it to a workspace. The `build` job fans-out into concurrently running test jobs: `integration-test`, `unit-test`, and `code-coverage`, each of which can have access to the jar by attaching the workspace.
```
Comment (thomasr8) [inline jekyll/_cci2/workspaces.md:15]: Question: Should this be ".jar file"?

**Review by thomasr8 — APPROVED** [review]
@rosieyohannan This is really good. I just had one small question.

File: jekyll/_cci2/workspaces.md [hunk 5]
```diff
@@ -88,6 +90,7 @@ For additional conceptual information on using workspaces, caching, and artifact
 Workspaces are stored for up to 15 days. Workspaces are not shared between pipelines, and the only way to access a workspace once the workflow has completed is if the workflow is rerun within the 15 day window.
 
 ## Workspaces and runner network charges
+{: #workspaces-and-runner-network-charges }
 
 When using self-hosted runners there is a network and storage usage limit included in your plan. Once your usage exceeds your limit charges will apply These charges are based on your accrued overages. The GB allocation for the plan you are on only applies to outbound traffic from CircleCI. Traffic within CircleCI is unlimited. Also, you will see no egress traffic or charges for any runners deployed within AWS `us-east-1`.
```
Comment (devinmarieb) [inline jekyll/_cci2/workspaces.md:94]: I think we're missing a period here.

```suggestion
When using self-hosted runners there is a network and storage usage limit included in your plan. Once your usage exceeds your limit charges will apply. These charges are based on your accrued overages.
```

File: jekyll/_cci2/workspaces.md (hunk 4 above)
Comment (rosieyohannan) [inline jekyll/_cci2/workspaces.md:15]:
```suggestion
For example, a project with a `build` job that builds a `.jar` file and saves it to a workspace. The `build` job fans-out into concurrently running test jobs: `integration-test`, `unit-test`, and `code-coverage`, each of which can have access to the jar by attaching the workspace.
```

**Pull request merged**